### PR TITLE
Bundle all client-side javascript in dashboard

### LINKED ISSUE #2768: Bundle all Javascript and CSS in dashboard
**Related**
#2671

**What would you like to be added**:
- [x] Replace scss-dass with vite
- [x] Add `main.js` to import all material-dashboard assets (in public/assets/)

**Why would you like this to be added**:
- Utilize our new shiny Vite compiler

### PR DESCRIPTION
**Pull request by Andrewnt219**
## Issue This PR Addresses
Closes #2768 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

**Changelog**
- Update Dockerfile to install dev deps as well
- Compile sass with vite
- Convert material-dashboard to .cjs (commonjs)
- Bundle all client-side js
- Move vite, npm-run-all and sass to dev deps.

## Steps to test the PR

- Go to status at /src/api/status
- Run `pnpm dev`
- Website is live at port 1111

Everything should be the same as [staging](https://dev.api.telescope.cdot.systems/v1/status/).

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: src/api/status/package.json [hunk 1]
```diff
@@ -9,9 +9,7 @@
     "start": "node src/server.js",
     "watch:server": "env-cmd -f env.local nodemon src/server.js",
     "compile:js": "vite build",
-    "watch:js": "vite build --watch",
-    "compile:scss": "sass ./scss/material-dashboard.scss:./public/assets/css/material-dashboard.css",
-    "watch:scss": "sass --watch ./scss/material-dashboard.scss:./public/assets/css/material-dashboard.css"
+    "watch:js": "vite build --watch"
```
Comment (DukeManh) [inline src/api/status/package.json:12]: As I understand this, we're using Vite to transform CommonJS modules into ESM and importing them. Why do need to watch for changes when `pnpm dev` when these dependencies are static?
```json
    "dev": "pnpm build && pnpm watch:server",
```

Comment (Andrewnt219) [reply src/api/status/package.json:12]: > As I understand this, we're using Vite to transform CommonJS modules into ESM and importing them. Why do need to watch for changes when pnpm dev when these dependencies are static?

Vite also builds our JS files (e.g. terminal.js). We don't want to manually run `vite build` on every js changes.

File: src/api/status/package.json [hunk 2]
```diff
@@ -23,12 +21,19 @@
...
+    "@popperjs/core": "2.10.2",
     "@senecacdot/satellite": "^1.x",
+    "bootstrap": "5.1.3",
     "express": "4.17.2",
     "express-handlebars": "6.0.2",
     "npm-run-all": "4.1.5",
+    "perfect-scrollbar": "1.5.1",
     "sass": "1.45.2",
-    "vite": "2.7.13"
+    "vite": "2.7.13",
```
Comment (humphd) [inline src/api/status/package.json:32]: Is this a dev dep?

**Review by humphd — COMMENTED** [review]
Looking really good, one question.

File: src/api/status/package.json (hunk 2 above)
Comment (Andrewnt219) [reply src/api/status/package.json:32]: Yes, technically. But when we need to run the build command in docker, which uses vite. Our docker only installs dependencies (which is a bit weird tbh).

Comment (humphd) [reply src/api/status/package.json:32]: We can change the dockerfile to do this in stages, see https://github.com/Seneca-CDOT/telescope/blob/master/src/api/sso/Dockerfile for an example of how to do it in stages, and install different things in different stages.

Comment (Andrewnt219) [reply src/api/status/package.json:32]: So I should change this line?

https://github.com/Seneca-CDOT/telescope/blob/39698783a296870d6655cd17202a30ea8df3f633/src/api/sso/Dockerfile#L32

Comment (humphd) [reply src/api/status/package.json:32]: As a temporary solution, sure.  But the real solution here is to do this in stages.  We need the dev dependencies to do the build, so we would install them in one layer, do the build, then copy the built files to another later that only installs the production dependencies.  We can help you write this in follow-ups.

cc @TDDR, @Meneguini, @Kevan-Y.

Comment (humphd) [inline src/api/status/package.json:32]: If you're going to remove that, might as well move this to dev deps too.

Comment (Andrewnt219) [reply src/api/status/package.json:32]: @manekenpix  should I change the docker file and install dev deps in prod as well?

Comment (humphd) [reply src/api/status/package.json:32]: I love that you don't trust me :)

It's fine. The only thing it does is add extra space to the built image.  But I can remove that with a mult-layer build later on.

Comment (Andrewnt219) [reply src/api/status/package.json:32]: How ironic, you are the only author of the file lol.

**Review by humphd — APPROVED** [review]

**Review by DukeManh — APPROVED** [review]
Amazing

**Pull request merged**